### PR TITLE
Update improvements

### PR DESCRIPTION
**Pull request by paul-ciobanu**

### CODE REVIEW

File: OctaneSdk/Services/EntityService.cs [hunk 1]
```diff
@@ -440,5 +450,20 @@ public CommitPattern ValidateCommitMessage(IRequestContext context, string commi
 
             return jsonSerializer.Deserialize<CommitPattern>(response.Data);
         }
+
+        /// <summary>
+        /// Return all possible transitions for a given entity type
+        /// </summary>
+        public async Task<EntityListResult<Transition>> GetPosibbleTransitionsForEntityType(IRequestContext context, string entityType)
```
Comment (radislavB) [inline OctaneSdk/Services/EntityService.cs]: 1.Errata - GetPosibbleTransitionsForEntityType =>GetPossibleTransitionsForEntityType
2.Why we need "Possible" in name of method, is it possible to call it GetTransitionsForEntityType ?

File: OctaneSdk/Entities/Base/BaseEntity.cs [hunk 2]
```diff
@@ -157,5 +157,14 @@ public override string ToString()
             String name = Name == null ? "No name" : Name.Substring(0, Math.Min(20, Name.Length)); //Show upto 20 characters in name
             return m_properties == null ? "No properties" : String.Format("{0} #{1} - {2} , {3} properties", TypeName, Id, name, m_properties.Count);
         }
+
+        public BaseEntity SimpleClone()
```
Comment (radislavB) [inline OctaneSdk/Entities/Base/BaseEntity.cs]: Its very vague name for Clone method. 
Implementation also is too custom.
Please think of something generic.

**Review by radislavB — CHANGES_REQUESTED** [review]
please see my comments

File: OctaneSdk/Services/EntityService.cs (hunk 1 above)
Comment (paul-ciobanu) [reply OctaneSdk/Services/EntityService.cs]: removed "possible" from the name

File: OctaneSdk/Entities/Base/BaseEntity.cs (hunk 2 above)
Comment (paul-ciobanu) [reply OctaneSdk/Entities/Base/BaseEntity.cs]: Removed clone method as I realized it was no longer needed

File: OctaneSdk/Entities/Base/Transition.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            get { return (Phase)GetValue(SOURCE_PHASE_FIELD); }
+        }
+
+        public Phase TargetPhase
+        {
+            get { return (Phase)GetValue(TARGET_PHASE_FIELD); }
+        }
+
+        public bool IsPrimary
+        {
```
Comment (radislavB) [inline OctaneSdk/Entities/Base/Transition.cs:36]: What will happen if GetValue will return null?
I think you should add method GetBooleanValue(fieldName, deaultValue) in DictionaryBasedEntity
, see example  GetStringValue

**Review by radislavB — CHANGES_REQUESTED** [review]
One more change required

File: OctaneSdk/Entities/Base/Transition.cs (hunk 3 above)
Comment (radislavB) [reply OctaneSdk/Entities/Base/Transition.cs:36]: Better, see implementation of GetLongValue (there are two overloads  - one return long? and second return just long but receive default value

Comment (paul-ciobanu) [reply OctaneSdk/Entities/Base/Transition.cs:36]: Added the 2 overloads like you suggested

**Review by radislavB — APPROVED** [review]
approved

**Pull request merged**